File: 14. Average Filtering/Laplacian.m
img = imread('moon.tif');
lap = fspecial('laplacian');
img_lap = imfilter(img, lap);
figure('Name', 'Original');
imshow(img);
figure('Name', 'Laplacian Blur');
imshow(img_lap, []);
figure('Name', 'Sharp');
imshow(img - img_lap);

File: 14. Average Filtering/Laplacian.py
import cv2
import numpy as np

img = cv2.imread('moon.tiff', 0)
k = cv2.Laplacian(img, -1)
sharp = cv2.subtract(img, k)
cv2.imshow("Original", img)
cv2.imshow("Laplacian", k)
cv2.imshow("Sharp", sharp)
